### PR TITLE
Closes #2555: Remove appservices gradle plugin, perform megazord substitution by hand.

### LINKED ISSUE #2555: Remove `org.mozilla.appservices` gradle plugin, perform megazord substitution by hand
As discussed in https://github.com/mozilla/application-services/issues/1097 the Application Services team would like to deprecate our special "megazording" gradle plugin, because it's complex to maintain and seems to interfere with other gradle features such as composite builds.

@ncalexan posted a proof-of-concept PR for how we might remove it and instead perform the megazord substitution by hand here:

https://github.com/mozilla-mobile/fenix/pull/1787

I've been poking at different variants of the approach and will put up a replacement PR shortly.

### PR DESCRIPTION
**Pull request by rfk**
Here's my current take on doing appservices megazord substitution by hand, ref https://github.com/mozilla-mobile/fenix/issues/2555.

Unlike the approach in https://github.com/mozilla-mobile/fenix/pull/1787, this doesn't explicitly list out all the individual appservices dependencies that should be replaced. Instead is assumes that **all** dependencies under `org.mozilla.appservices` should be replaced by the configured all-in-one megazord.

This seems to me both simpler, and more likely to fail early and noisily if something goes wrong. For example, if this project takes a new (possibly transitive) dependency on another appservices component but we forget to add it to the `fenix-megazord` build, then the approach here will result in a build failure; I tested this by trying to build with `lockbox-megazord` instead. That might be jarring for the developer who introduced the new dependency and gets an unexpected build failure, but seems better than accidentally shipping a mix of megazorded and non-megazorded appservices components.

With this change, I have had some tentative success in doing a local composite build and substituting in local versions of appservices components, but I think there are still a few bugs to shake out on that front, so won't make any promises just yet. But it should be a good step in that direction.

/cc @ncalexan @colintheshots @thomcc 

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] ~~**Tests**: This PR includes thorough tests or an explanation of why it does not~~
  - I couldn't think of a way to test this other than confirming that existing tests still work
- [ ] ~~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one~~
  - I didn't add one because it seemed like the existing entries were user-facing changes rather than infrastructural things, but I'm happy to add one.
- [ ] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~
    - No user-facing features

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -378,7 +367,6 @@ dependencies {
     testImplementation Deps.junit
     testImplementation Deps.robolectric
     implementation Deps.fragment_testing
-    testImplementation Deps.megazord_forUnitTests
```
Comment (rfk) [inline app/build.gradle]: IMHO you shouldn't need to list this explicitly as a test dependency in the same way you don't list it explicitly as a build dependency.

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -28,9 +28,8 @@ private object Versions {
     const val androidx_core = "1.2.0-alpha01"
     const val androidx_transition = "1.1.0-rc01"
 
-    const val appservices_gradle_plugin = "0.4.4"
     const val mozilla_android_components = "0.53.0-SNAPSHOT"
-    const val mozilla_appservices = "0.27.0"
+    const val mozilla_appservices = "0.27.1"
```
Comment (rfk) [inline buildSrc/src/main/java/Dependencies.kt]: This got out of sync with the version used in android-components; if you like I could add some more logic to assert that we only see a single version number among any appservices dependencies.

File: app/build.gradle [hunk 3]
```diff
@@ -416,4 +404,31 @@ task printGeckoviewVersions {
 
 // Normally this should use the same version as the glean dependency. But since we are currently using AC snapshots we
 // can't reference a git tag with a specific version here. So we are just using "master" and hoping for the best.
-apply from: 'https://github.com/mozilla-mobile/android-components/raw/master/components/service/glean/scripts/sdk_generator.gradle'
\ No newline at end of file
+apply from: 'https://github.com/mozilla-mobile/android-components/raw/master/components/service/glean/scripts/sdk_generator.gradle'
+
+// For production builds, the native code for all `org.mozilla.appservices` dependencies gets compiled together
+// into a single "megazord" build, and different megazords are published for different subsets of features.
+// Ref https://mozilla.github.io/application-services/docs/applications/consuming-megazord-libraries.html
```
Comment (rfk) [inline app/build.gradle]: I haven't updated this doc yet but I will, based on any feedback here.

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (grigoryk) [reply buildSrc/src/main/java/Dependencies.kt]: This got re-added with https://github.com/mozilla-mobile/fenix/pull/2282/files, and afaik only used for roboelectric tests. It'll be good to both comment on this fact here, and ensure that we don't diverge (if that's possible!)

**Comment by rfk** [comment]
@grigoryk I wonder if it would make sense for a-c to export some sort of "forUnitTest" dependency wrapper itself, so that fenix doesn't have to take a direct dependency here. Not sure what that would look like in practice though.

**Comment by rfk** [comment]
@grigoryk updated with your suggestions, this will now error out if it detects multiple versions of `org.mozilla.appservices` packages in a single build.

**Comment by boek** [comment]
@grigoryk want to give this a quick look again before we land?

**Comment by rfk** [comment]
Rebased for merge conflicts

**Pull request merged**